persistence/src/test/java/cz/muni/fi/pa165/mushrooms/dao/ForestDaoTest.java

```diff
@@ -0,0 +1,157 @@
+package cz.muni.fi.pa165.mushrooms.dao;
+
+import cz.muni.fi.pa165.mushrooms.entity.Forest;
+import cz.muni.fi.pa165.mushrooms.validation.PersistenceSampleApplicationContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.orm.jpa.JpaSystemException;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.AbstractTransactionalJUnit4SpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.validation.ConstraintViolationException;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * Created by Matúš on 28.10.2017.
+ */
+@ContextConfiguration(classes = PersistenceSampleApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+public class ForestDaoTest extends AbstractTransactionalJUnit4SpringContextTests {
+
+    @Autowired
+    private ForestDao forestDao;
+
+    @PersistenceContext
+    private EntityManager em;
+
+    private Forest forest;
+
+    @Before
+    public void setUp(){
+        forest = new Forest();
+        forest.setName("Dun Morogh");
+        forest.setDescription("Dun Morogh forest in South Wisconsin");
+
+        em.persist(forest);
+        em.flush();
+    }
+
+    @Test
+    public void findById_validId() throws Exception {
+        Forest testForest = forestDao.findById(forest.getId());
+        assertThat(testForest).isNotNull();
+        assertThat(testForest).isEqualToComparingFieldByField(forest);
+    }
+
+    @Test
+    public void findById_nullId() throws Exception {
+        assertThatThrownBy(() -> forestDao.findById(null)).hasRootCauseExactlyInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void create_validForest() throws Exception {
+        int sizeBeforeCreate = forestDao.findAll().size();
+        Forest testForest = new Forest();
+        testForest.setName("Duskwood");
+
+        forestDao.create(testForest);
+        Forest foundForest = em.find(Forest.class, testForest.getId());
+
+        assertThat(foundForest).isNotNull();
+        assertThat(forestDao.findAll().size()).isEqualTo(sizeBeforeCreate + 1);
```

Comment:
assert the collection contains the expected elements (AssertJ has a nice method)